src/components/dialogs/StandardDialog.svelte

```diff
@@ -5,68 +5,99 @@
-  }
-  let previousExpanded = false;
-  dialog.subscribe(({ expanded }) => {
-    if (previousExpanded && !expanded) {
-      onClose();
-      if (finalFocusRef) {
-        (finalFocusRef as HTMLElement).focus();
-      }
+  };
+
+  const onCloseDialog = () => {
+    if (finalFocusRef) {
+      (finalFocusRef as HTMLElement).focus();
+    }
+    onClose();
+  };
+
+  const onOpenChange: CreateDialogProps['onOpenChange'] = ({ next }) => {
+    if (!next) {
+      onCloseDialog();
+    } else {
+      onOpenDialog();
     }
-    previousExpanded = expanded;
+    return next;
+  };
+
+  const {
+    elements: { overlay, content, title: titleElement, close, portalled },
+    states,
+  } = createDialog({
+    forceVisible: true,
+    preventScroll: true,
+    closeOnOutsideClick: true,
+    closeOnEscape: true,
+    onOpenChange,
+    ...createDialogPropsOverride,
   });
+
+  const { open } = states;
+
+  // Syncing inside and outside component states to minimise prop changes
+  const sync = createSync(states);
+  $: sync.open(isOpen, v => (isOpen = v));
+
+  $: if (isOpen) {
+    onOpenDialog();
+  } else {
+    onCloseDialog();
+  }
 </script>
 
-<div class="fixed z-10">
-  <Transition show={$dialog.expanded}>
-    <!-- svelte-ignore a11y-no-static-element-interactions -->
-    <!-- Keyboard event handler for on:click is implemented as part of svelte-headlessui dialog builder  -->
+<div class="fixed z-10" use:melt={$portalled}>
+  {#if $open}
     <div
       class="fixed top-0 left-0 h-screen w-screen flex justify-center items-center bg-black/50 bg-blend-darken"
-      use:dialog.modal
-      on:click={dismissOnClickOutside ? dialog.close : undefined}>
-      <Transition
-        enter="transition ease-out duration-200"
-        enterFrom="transform opacity-0 scale-95"
-        enterTo="transform opacity-100 scale-100"
-        leave="transition ease-in duration-75"
-        leaveFrom="transform opacity-100 scale-100"
-        leaveTo="transform opacity-0 scale-95">
-        <!-- on:click for stopping propogation  -->
-        <div
-          class="w-min h-min border-gray-200 border border-solid relative bg-white rounded-lg p-8 z-15"
-          on:click|stopPropagation>
-          {#if hasCloseButton}
-            <div class="absolute right-2 top-2">
-              <IconButton ariaLabel={$t('actions.close')} onClick={dialog.close}>
-                <CloseIcon class="text-xl m-1" />
-              </IconButton>
-            </div>
+      use:melt={$overlay}
+      transition:fade={{ duration: 100 }}>
+      <div
+        use:melt={$content}
+        class="w-min h-min border-gray-200 border border-solid relative bg-white rounded-lg p-8 z-15"
+        transition:scale={{
+          duration: 200,
+          start: 0.9,
+          easing: quintOut,
+        }}>
+        {#if hasCloseButton}
+          <div class="absolute right-2 top-2">
+            <IconButton
+              {...$close}
+              useAction={$close.action}
```

Comment:
Can this just be `onClick={onClose}`, then we can revert the type variant change to IconButton?